tools/realsense-viewer/auto_updater/update_handler.cpp

```diff
@@ -0,0 +1,271 @@
+                            cvt_ok = false;
+                    }
+                    else if (element_key == "version")
+                    {
+                        version_info.version = it.value().get<long long>();
+                        json_schema_verifier.version_cnt++;
+                    }
+                    else if (element_key == "platform")
+                    {
+                        if (verify_platform(it.value().get<std::string>()))
+                        {
+                            version_info.platform = it.value().get<std::string>();
+                            json_schema_verifier.platform_cnt++;
+                        }
+                    }
+                    else if (element_key == "link")
+                    {
+                        version_info.link = it.value().get<std::string>();
+                        json_schema_verifier.link_cnt++;
+                    }
+                    else
+                    {
+                        cvt_ok = false;
+                    }
+
+                    if (!cvt_ok)
+                    {
+                        throw std::runtime_error("Server versions json file corrupted\n");
+                    }
+                }
+                if (json_schema_verifier.validate())
+                {
+                    _server_versions_vec.emplace_back(version_info);
+                    json_schema_verifier.reset();
+                }
+                else
+                {
+                    throw std::runtime_error("Server versions json file corrupted\n");
+                }
+            }
+        }
+        else
+        {
+            throw std::runtime_error("Server versions json file corrupted - Title mot as expected\n");
+        }
+    }
+
+    const std::string update_handler::get_platform() const
+    {
+        std::string platform;
+
+#ifdef _WIN64
+        platform = "Windows amd64";
+
+#elif _WIN32
+        platform = "Windows x86";
+#elif __linux__ 
+#ifdef __arm__
+        platform = "Linux arm";
+#else
+        platform = "Linux amd64";
+#endif
+#elif __APPLE__
+        platform = "Mac OS";
+#elif __ANDROID__
+        //TBD
+#endif
+        return platform;
+    }
+
+    bool update_handler::verify_platform(const std::string& pl) const
+    {
+        std::unordered_set<std::string> pl_set;
+        pl_set.emplace("Windows amd64");
+        pl_set.emplace("Windows x86");
+        pl_set.emplace("Linux amd64");
+        pl_set.emplace("Linux arm");
+        pl_set.emplace("Mac OS");
+
+        auto val = pl_set.find(pl);
+        if (val != pl_set.end())
+        {
+            return true;
+        }
+        return false;
+    }
+    bool update_handler::load_server_versions()
+    {
+        std::stringstream server_versions_data;
+        if (!get_server_data(server_versions_data))
+        {
+            return false; // Failed to get version from server
+        }
+        else
+        {
+            try
```

Comment:
we should not have throw and catch in same class

Reply:
Will be fixed